src/compiler/parser.ts

```diff
@@ -4545,7 +4545,18 @@ namespace ts {
                         parseErrorAtRange(openingTag, Diagnostics.JSX_fragment_has_no_corresponding_closing_tag);
                     }
                     else {
-                        parseErrorAtRange(openingTag.tagName, Diagnostics.JSX_element_0_has_no_corresponding_closing_tag, getTextOfNodeFromSourceText(sourceText, openingTag.tagName));
+                        // We want the error to span only property identifier "Foo.Bar" e.g in < Foo.Bar >...
+                        const tag = openingTag.tagName as any;
+                        let start = tag.pos;

```

Comment:
I think a better/simpler way would be to just write

```ts
const start = skipTrivia(sourceText, openingTag.tagName.pos);
```

Reply:
I have changed it to use skipTrivia instead. Thanks a lot for the review.